File: QadeerApp.Web/Modules/Administration/Specialization/SpecializationColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.Specialization")]
[BasedOnRow(typeof(SpecializationRow), CheckNames = true)]
public class SpecializationColumns
{
    [EditLink, AlignRight, Width(60)]
    public int SpecializationId { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(200)]
    public string DepartmentName { get; set; }

    [Width(80), QuickFilter]
    public short IsActive { get; set; }
}

File: QadeerApp.Web/Modules/ServerTypes/Administration/SpecializationColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SpecializationRow } from "./SpecializationRow";

export interface SpecializationColumns {
    SpecializationId: Column<SpecializationRow>;
    Name: Column<SpecializationRow>;
    DepartmentName: Column<SpecializationRow>;
    IsActive: Column<SpecializationRow>;
}

export class SpecializationColumns extends ColumnsBase<SpecializationRow> {
    static readonly columnsKey = 'Administration.Specialization';
    static readonly Fields = fieldsProxy<SpecializationColumns>();
}
